src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,24 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
     private final Executor executor;
+    private final Map<String, UserData> cache = new ConcurrentHashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        if (cache.containsKey(userId)) {
+            return CompletableFuture.supplyAsync(() -> cache.get(userId), executor);
+        } else {
+            cache.put(userId, new UserData(userId, "Details for " + userId));
+        }
+        return CompletableFuture.supplyAsync(() -> cache.get(userId), executor);
     }
```

Comment:
There is a potential race condition here. If two threads call `processRequest` with the same `userId` at the same time, they could both pass the `containsKey` check before either puts the new `UserData` into the cache, leading to `UserData` being created twice for the same `userId`. To resolve this issue, consider using `computeIfAbsent` method of the `ConcurrentHashMap` which atomically computes the value for a key if it's not present.